src/redis-benchmark.c

```diff
@@ -846,6 +846,10 @@ static client createClient(char *cmd, size_t len, client from, int thread_id) {
     }
     if (config.idlemode == 0)
         aeCreateFileEvent(el,c->context->fd,AE_WRITABLE,writeHandler,c);
+    else
+        /*in idle mode, clients still need to register readHandler for catching error */
+        aeCreateFileEvent(el,c->context->fd,AE_WRITABLE,readHandler,c);
```

Comment:
you register the **read** handler for the WRITABLE event.
i suppose that's a typo?

maybe it's possible to just drop the `if` and register always register the WRITABLE event?